### PR TITLE
Removes bingo for `go run`

### LINKED ISSUE #314: replace bingo with go 1.17 `go run` commands
**Is your feature request related to a problem? Please describe.**
Right now, we use a modified version of bingo to pin versions without tainting GOBIN's default. For example, this is used to invoke known versions of hugo, goreleaser, golint, etc

**Describe the solution you'd like**
Inside the Makefile, declare paths for each main tool and do `go run` instead. This reduces the toolchain knowledge to run the project and conversion to make-unfriendly env like windows more straightforward.

> go run now accepts arguments with version suffixes (for example, go run example.com/cmd@v1.0.0). This causes go run to build and run packages in module-aware mode, ignoring the go.mod file in the current directory or any parent directory, if there is one. This is useful for running executables without installing them or without changing dependencies of the current module. 

**Describe alternatives you've considered**
Leaving it alone. There is some benefit to leaving it as depending on how `go run` in 1.17 is implemented, it might be inefficient to repeatedly run the same command.

**Additional context**
https://tip.golang.org/doc/go1.17#go-command

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This uses versioned syntax of `go run` brought to us in 1.17
https://golang.org/doc/go1.17

In doing so, we can remove the complexity of bingo and also simplify its
impact on windows.

Fixes #314

### CODE REVIEW

File: .github/workflows/msi.yaml [hunk 1]
```diff
@@ -72,9 +72,8 @@ jobs:
         with:
           path: |  # TODO: go build cache if we care, noting it is OS-specific
```
Comment (codefromthecrypt) [inline .github/workflows/msi.yaml:73]: I will sort out the build cache in a different PR as it requires handling Os-specifics!

**Review by mathetake — APPROVED** [review]
nice :smile:

**Comment by codefromthecrypt** [comment]
might have a glitch on goreleaser on windows. regardless will be green prior to merge!

```
± |no-bingo {1} ✓| → make clean msi
--- clean ---
--- win-bin ---
GOOS=windows GOARCH=amd64 go run github.com/goreleaser/goreleaser@v0.175.0 build --snapshot --single-target --rm-dist
fork/exec /var/folders/vd/1cf8zdb1721f4z5rjggy8bp40000gn/T/go-build2241750511/b001/exe/goreleaser.exe: exec format error
make: *** [dist/func-e_windows_amd64/func-e.exe] Error 1
```

**Comment by codefromthecrypt** [comment]
note: this is slower than before because go rebuilds each time. However, this time spent is less important to me than lack of cruft, so think it is a better tradeoff. @mathetake wdyt?

File: Makefile [hunk 2]
```diff
@@ -123,24 +128,20 @@ coverage:
...
-format: $(GOIMPORTS) ## Format all Go code
+format: ## Format all Go code
 	@echo "--- format ---"
-	@$(LICENSER) apply -r "Tetrate"
+	@go run $(licenser) apply -r "Tetrate"
 	@find . -type f -name '*.go' | xargs gofmt -s -w
-	@for f in `find . -name '*.go'`; do \
-	    awk '/^import \($$/,/^\)$$/{if($$0=="")next}{print}' $$f > /tmp/fmt; \
-	    mv /tmp/fmt $$f; \
-	    $(GOIMPORTS) -w -local github.com/tetratelabs/func-e $$f; \
```
Comment (codefromthecrypt) [inline Makefile:142]: doing this in a loop is as time consuming as trying to parse wtf ^^ was trying to do

File: Makefile [hunk 3]
```diff
@@ -43,9 +46,11 @@ MSI_VERSION ?= 0.0.1
 # Right now, the only arch we support is amd64 because Envoy doesn't yet support arm64 on Windows
 # https://github.com/envoyproxy/envoy/issues/17572
 # Once that occurs, we will need to set -arch arm64 and bundle accordingly.
-$(WIN_BIN_EXE): $(GORELEASER)
+$(WIN_BIN_EXE):
 	@echo "--- win-bin ---"
-	@GOOS=windows GOARCH=amd64 $(GORELEASER) build --snapshot --single-target --rm-dist
+    # We can't pass GOOS=windows GOARCH=amd64 as it ends up interpreted by go, not goreleaser.
```
Comment (codefromthecrypt) [inline Makefile:51]: this is annoying to build all when you only need windows, but I think we will be killing goreleaser soon anyway

File: Makefile [hunk 4]
```diff
@@ -43,9 +46,11 @@ MSI_VERSION ?= 0.0.1
...
+    # Hence, we build all targets even if we only need Windows.
+	@go run $(goreleaser) build --snapshot --rm-dist
```
Comment (mathetake) [inline Makefile:53]: I like being better aligned 😄 

```suggestion
	# We can't pass GOOS=windows GOARCH=amd64 as it ends up interpreted by go, not goreleaser.
	# Hence, we build all targets even if we only need Windows.
	@go run $(goreleaser) build --snapshot --rm-dist
```

**Comment by codefromthecrypt** [comment]
thanks takeshi-san!

**Pull request merged**